lib/ansible/modules/cloud/vmware/vmware_dvs_portgroup.py

```diff
@@ -157,242 +217,1038 @@
+                message = changed_list[0]
+            message += changed_suffix
+            if not self.module.check_mode:
+                try:
+                    task = self.dvs_portgroup.ReconfigureDVPortgroup_Task(pg_spec)
+                    wait_for_task(task)
+                except TaskError as invalid_argument:
+                    self.module.fail_json(msg="Failed to update portgroup : %s" % to_native(invalid_argument))
+        else:
+            message = "Portgroup already configured properly"
+        results['changed'] = changed
+        results['result'] = message
+
+        self.module.exit_json(**results)
 
 
 def main():
+    """Main"""
     argument_spec = vmware_argument_spec()
     argument_spec.update(
         dict(
-            portgroup_name=dict(required=True, type='str'),
-            switch_name=dict(required=True, type='str'),
-            vlan_id=dict(required=True, type='str'),
-            num_ports=dict(required=True, type='int'),
-            portgroup_type=dict(required=True, choices=['earlyBinding', 'lateBinding', 'ephemeral'], type='str'),
-            state=dict(required=True, choices=['present', 'absent'], type='str'),
-            vlan_trunk=dict(type='bool', default=False),
-            network_policy=dict(
+            switch_name=dict(type='str', required=True, aliases=['switch']),
+            portgroup=dict(type='str', required=True, aliases=['portgroup_name']),
+            description=dict(type='str', aliases=['portgroup_description']),
+            vlan_id=dict(type='int', default=0),
+            vlan_trunk_range=dict(type='list'),
+            private_vlan_id=dict(type='int'),
+            num_ports=dict(type='int', aliases=['ports']),
+            network_resource_pool=dict(type='str', default='default'),
+            port_binding=dict(type='str', default='static', choices=['static', 'dynamic', 'ephemeral']),
+            port_allocation=dict(type='str', choices=['fixed', 'elastic']),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            security=dict(
                 type='dict',
                 options=dict(
-                    promiscuous=dict(type='bool', default=False),
+                    promiscuous_mode=dict(type='bool', default=False),
                     forged_transmits=dict(type='bool', default=False),
                     mac_changes=dict(type='bool', default=False)
                 ),
                 default=dict(
-                    promiscuous=False,
+                    promiscuous_mode=False,
                     forged_transmits=False,
                     mac_changes=False
-                )
+                ),
+                aliases=['network_policy', 'security_policy']
             ),
-            teaming_policy=dict(
+            teaming=dict(
                 type='dict',
                 options=dict(
-                    inbound_policy=dict(type='bool', default=False),
+                    load_balancing=dict(
```

Comment:
Suggest to add the alias for `load_balancing_policy` for backward compatibility, including in doc

Reply:
done